src/freedv_interface.cpp

```diff
@@ -122,10 +144,29 @@ void FreeDVInterface::setRunTimeOptions(int clip, int bpf, int phaseEstBW, int p
 {
     for (auto& dv : dvObjects_)
     {
-        freedv_set_clip(dv, clip);   // 700C/700D/700E/2020
-        freedv_set_tx_bpf(dv, bpf);  // 700D/700E
-        freedv_set_phase_est_bandwidth_mode(dv, phaseEstBW); // 700D/2020
-        freedv_set_dpsk(dv, phaseEstDPSK);  // 700D/2020
+        switch(freedv_get_mode(dv))
```

Comment:
I think the original implementation above is OK and less complicated.  Take a look at the implementation of these functions in https://github.com/drowe67/codec2/blob/master/src/freedv_api.c - they don't do anything unless they are called in the relevant mode.  If that's not the case, I'm happy to tweak `freedv_api.c`

Reply:
Reverted back to the prior implementation.